librz/analysis/data.c

```diff
@@ -2,36 +2,26 @@
 // SPDX-License-Identifier: LGPL-3.0-only
 
 #include <rz_analysis.h>
+#include <rz_util.h>
 
-#define MINLEN 1
-static int is_string(const ut8 *buf, int size, int *len) {
-	int i;
-	if (size < 1) {
-		return 0;
+static bool get_string(const ut8 *buf, int size, RzDetectedString **dstr, RzStrEnc encoding) {
+	if (!buf || size < 1) {
+		return false;
 	}
-	if (size > 3 && buf[0] && !buf[1] && buf[2] && !buf[3]) {
-		*len = 1; // XXX: TODO: Measure wide string length
-		return 2; // is wide
-	}
-	for (i = 0; i < size; i++) {
-		if (!buf[i] && i > MINLEN) {
-			*len = i;
-			return 1;
-		}
-		if (buf[i] == 10 || buf[i] == 13 || buf[i] == 9) {
-			continue;
-		}
-		if (buf[i] < 32 || buf[i] > 127) {
-			// not ascii text
-			return 0;
-		}
-		if (!IS_PRINTABLE(buf[i])) {
-			*len = i;
-			return 0;
-		}
+
+	RzUtilStrScanOptions opt = {
+		.buf_size = size,
+		.max_uni_blocks = 4,
+		.min_str_length = 4,
+		.prefer_big_endian = false,
```

Comment:
Shouldn't we use endianness from the RzAnalysis here?

Reply:
hmmm probably